cli/src/cmd_net.rs

```diff
@@ -0,0 +1,828 @@
+use clap::Parser;
+use colored::*;
+use oxide::{
+    context::Context,
+    types::{
+        Address, AddressConfig, BgpPeerConfig, BgpPeerStatus, IpNet, LinkConfigCreate,
+        LldpServiceConfigCreate, NameOrId, Route, RouteConfig, SwitchInterfaceConfigCreate,
+        SwitchInterfaceKind, SwitchInterfaceKind2, SwitchLocation, SwitchPort,
+        SwitchPortConfigCreate, SwitchPortGeometry, SwitchPortGeometry2, SwitchPortSettingsCreate,
+    },
+    Client, ClientSystemHardwareExt, ClientSystemNetworkingExt,
+};
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+use tabwriter::TabWriter;
+use uuid::Uuid;
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "net")]
+pub struct CmdNet {
+    #[clap(subcommand)]
+    subcmd: NetSubCommand,
+}
+
+#[derive(Parser, Debug, Clone)]
+enum NetSubCommand {
+    Addr(CmdAddr),
+    Port(CmdPort),
+    Bgp(CmdBgp),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "port")]
+pub struct CmdPort {
+    #[clap(subcommand)]
+    subcmd: PortSubCommand,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdPort {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        match &self.subcmd {
+            PortSubCommand::Config(cmd) => cmd.run(ctx).await,
+            PortSubCommand::Status(cmd) => cmd.run(ctx).await,
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+enum PortSubCommand {
+    Config(CmdPortConfig),
+    Status(CmdPortStatus),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "bgp")]
+pub struct CmdBgp {
+    #[clap(subcommand)]
+    subcmd: BgpSubCommand,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdBgp {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        match &self.subcmd {
+            BgpSubCommand::Status(cmd) => cmd.run(ctx).await,
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+enum BgpSubCommand {
+    Status(CmdBgpStatus),
+}
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "addr")]
+pub struct CmdAddr {
+    #[clap(subcommand)]
+    subcmd: AddrSubCommand,
+}
+
+#[derive(Parser, Debug, Clone)]
+enum AddrSubCommand {
+    Add(CmdAddrAdd),
+}
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+enum Switch {
+    Switch0,
+    Switch1,
+}
+
+impl std::fmt::Display for Switch {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", format!("{self:?}").to_lowercase())
+    }
+}
+
+#[derive(clap::ValueEnum, Clone, Debug)]
+enum Port {
+    Qsfp0,
+    Qsfp1,
+    Qsfp2,
+    Qsfp3,
+    Qsfp4,
+    Qsfp5,
+    Qsfp6,
+    Qsfp7,
+    Qsfp8,
+    Qsfp9,
+    Qsfp10,
+    Qsfp11,
+    Qsfp12,
+    Qsfp13,
+    Qsfp14,
+    Qsfp15,
+    Qsfp16,
+    Qsfp17,
+    Qsfp18,
+    Qsfp19,
+    Qsfp20,
+    Qsfp21,
+    Qsfp22,
+    Qsfp23,
+    Qsfp24,
+    Qsfp25,
+    Qsfp26,
+    Qsfp27,
+    Qsfp28,
+    Qsfp29,
+    Qsfp30,
+    Qsfp31,
+}
```

Comment:
I wonder if this is something that could live in the API? Lists like this could be difficult to maintain the same in various clients :thinking: 

Additionally, if our customers were to create their own clients from our API, they might find it pretty hard to know that this enum is required

Reply:
The network configuration API is going to be changing pretty significantly based on some work @internet-diglett  is doing to eventually support things like terraform integration. A resource-based model is ultimately where we want to go. This PR is to make things reasonable for operators until that larger chunk of API work lands. It's my hope that at a CLI level, these commands will remain pretty much the same, while their internal implementation will benefit from broader API work, and things like this, which you rightfully note should be captured by the API will go away.

Reply:
gotcha, thanks for the clarification!